analyzers/dataframe/MCParticle.cc

```diff
@@ -366,3 +366,20 @@ bool  filterMC_pdgID::operator() (ROOT::VecOps::RVec<edm4hep::MCParticleData> in
   }
   return false;
 }
+
+
+TVector3 getMC_EventPrimaryVertex( ROOT::VecOps::RVec<edm4hep::MCParticleData> in )  {
```

Comment:
why don't you add as an argument the generator status? as you use it later on, the function can be generalised, with as default value 21 for the status

Reply:
Salut Clement ! you mean, in case some generators use a different convention, or in case the Pythia8 status of the incoming particles is changed in the future ? yes, that's a good suggestion.

Reply:
yes, precisely. I try to avoid having hard-coded numbers